src/vs/workbench/contrib/terminal/electron-browser/terminalInstance.ts

```diff
@@ -968,8 +987,13 @@ export class TerminalInstance implements ITerminalInstance {
 		}
 	}
 
-	private _onProcessExit(exitCode: number): void {
-		this._logService.debug(`Terminal process exit (id: ${this.id}) with code ${exitCode}`);
+	private _onProcessOrExtensionCallbackExit(exitCode?: number): void {
```

Comment:
Can you add a jsdoc comment here stating that exitCode is undefined iff the terminal was disposed through user action.